### PR TITLE
feat: register account repository (AR-1066)

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

AR-1066

### Testing

unit tests

----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by ohassine — APPROVED** [review]
👌

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/register/RegisterAccountRepository.kt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import com.wire.kalium.logic.NetworkFailure
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.wrapApiRequest
+import com.wire.kalium.network.api.user.register.RegisterApi
+import com.wire.kalium.network.api.user.register.RegisterResponse
+
+interface RegisterAccountRepository {
+    suspend fun requestEmailActivationCode(email: String, baseApiHost: String): Either<NetworkFailure, Unit>
+    suspend fun verifyActivationCode(email: String, code: String, baseApiHost: String): Either<NetworkFailure, Unit>
```
Comment (mchenani) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/register/RegisterAccountRepository.kt:11]: What do you think of a more general name like `verfiyAccount`? In my opinion that would make it more readable.
`verifyAccoutn(email, verificationCode....`

**Review by mchenani — APPROVED** [review]
🥁✌️

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/register/RegisterAccountRepository.kt (hunk 1 above)
Comment (MohamadJaara) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/register/RegisterAccountRepository.kt:11]: `verifyAccount` implies that there is an account which it's not the case

**Pull request merged**